arbnode/espresso/chain/delay_chain.go

```diff
@@ -0,0 +1,58 @@
+package chain
+
+import (
+	"context"
+	"time"
+
+	espresso_client "github.com/EspressoSystems/espresso-network/sdks/go/client"
+	espresso_types "github.com/EspressoSystems/espresso-network/sdks/go/types"
+	espresso_common "github.com/EspressoSystems/espresso-network/sdks/go/types/common"
+	"github.com/offchainlabs/nitro/arbnode/espresso"
+)
+
+// EspressoChainDelayed is a wrapper around an Espresso chain client that
+// introduces a delay before executing any method calls. This can be useful for
+// simulating network latency or for testing purposes.
```

Comment:
nit: In arbitrum all the testing helper code goes inside the system_tests folder. I think we should put this file there as well